### PR TITLE
vendor: switch to godep and update `gogo/protobuf`

### PR DESCRIPTION
**Pull request by nolouch**
update `gogo/protobuf` to `v1.1.1` and also ` grpc` to `1.12.0`

### CODE REVIEW

**Comment by overvenus** [comment]
How do TiDB and PD build against this PR? Will it cause binary bloat?

File: Makefile [hunk 1]
```diff
@@ -4,9 +4,6 @@ CURDIR := $(shell pwd)
 
 KEEP_FILE := '**/*.proto,**/*.sh'
```
Comment (overvenus) [inline Makefile]: It can be removed.

File: Gopkg.toml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+[[constraint]]
+  name = "github.com/golang/protobuf"
+  source = "https://github.com/golang/protobuf.git"
+  version = "1.2.0"
+
+[[constraint]]
+  branch = "master"
+  name = "golang.org/x/net"
+
+[prune]
```
Comment (overvenus) [inline Gopkg.toml]: How about pruning `non-go = true`?

File: Makefile [hunk 3]
```diff
@@ -4,9 +4,6 @@ CURDIR := $(shell pwd)
...
 
-export PATH := $(CURDIR)/_vendor/bin:$(PATH)
-export GOPATH := $(CURDIR)/_vendor
```
Comment (overvenus) [inline Makefile]: By removing these lines, generate_go may install few go tools into your real gopath and use wrong protobuf and grpc compiler.

File: Gopkg.toml (hunk 2 above)
Comment (nolouch) [reply Gopkg.toml]: some licenses should be kept.

**Comment by zhexuany** [comment]
dep part looks good to me.

**Comment by nolouch** [comment]
@overvenus I test `tidb` with this change and no binary bloat.

**Comment by nolouch** [comment]
close by: https://github.com/pingcap/kvproto/pull/300 
new pr will trigger the ci.

**Pull request closed without merging**